flink-streaming-java/src/main/java/org/apache/flink/streaming/api/environment/RemoteStreamEnvironment.java

```diff
@@ -192,138 +159,46 @@ public RemoteStreamEnvironment(String host, int port, Configuration clientConfig
 			try {
 				URL jarFileUrl = new File(jarFile).getAbsoluteFile().toURI().toURL();
 				this.jarFiles.add(jarFileUrl);
-				ClientUtils.checkJarFile(jarFileUrl);
+				JarUtils.checkJarFile(jarFileUrl);
 			} catch (MalformedURLException e) {
 				throw new IllegalArgumentException("JAR file path is invalid '" + jarFile + "'", e);
 			} catch (IOException e) {
 				throw new RuntimeException("Problem with jar file " + jarFile, e);
 			}
 		}
+
 		if (globalClasspaths == null) {
 			this.globalClasspaths = Collections.emptyList();
-		}
-		else {
+		} else {
 			this.globalClasspaths = Arrays.asList(globalClasspaths);
 		}
-		this.savepointRestoreSettings = savepointRestoreSettings;
-	}
 
-	/**
```

Comment:
I don't like this set of methods but I also think that we shouldn't remove them in this change.

Reply:
Reverted.

Reply:
Do you ask for doing so in a separate story?

Reply:
Maybe, but they were added for a purpose. Check out the Jira issue for the changes.

Reply:
OK.